cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/BerryClient.scala

```diff
@@ -123,7 +123,7 @@ class BerryClient(val jsonParser: JSONParser,
 
       if (nextInterval.toDurationMillis <= 0 || hasEnoughResults(mergedResults, targetLimits)) {
         val returnedResult =
-          if (mergedResults.head.value.size > targetLimits) Seq(JsArray(mergedResults.head.value.take(targetLimits)))
+          if (targetLimits.nonEmpty && mergedResults.head.value.size > targetLimits.get) Seq(JsArray(mergedResults.head.value.take(targetLimits.get)))
```

Comment:
`take` function can return less result if there are not enough results inside the list. So another easier way of writing the code is 
```
val limitResultOpt =targetLimits.map(limit => Seq(JsArray(mergedResults.head.value.take(limit))))
val returnedResult = limitResultOpt.getOrElse(mergedResult)
```

which is cleaner than the `if else` clause.